### ISSUE TITLE
netifaces.ifaddresses causes value error for certain interfaces

### ISSUE THREAD

**Issue by VincentKetelaars** [Tribler/dispersy#171]
In dispersy.py all interfaces will be retrieved by netifaces.interfaces(), but it can not handle of these..

  File "/home/vincent/git/dispersy-experiments/dispersy/dispersy.py", line 334, in **init**
    self._local_interfaces = list(self._get_interface_addresses())
  File "/home/vincent/git/dispersy-experiments/dispersy/dispersy.py", line 409, in _get_interface_addresses
    addresses = netifaces.ifaddresses(interface)
ValueError: You must specify a valid interface name.

My particular interface was "ppp0".

**Comment by boudewijn-tribler** [comment]
#181 ensures that the ppp0 interface is ignored without crashing.  Do you need to use the ppp0 interface, or is ignoring it sufficient for you right now?

**Comment by VincentKetelaars** [comment]
You should do whatever you think best. I am using ppp interfaces, and I was using that Dispersy function to check if the interface exists. But I can work around that.

**Comment by boudewijn-tribler** [comment]
The best thing would be to see why netifaces barfs on your ppp interface.  I assume you are using netifaces 0.8?

A quick search shows that there were issues with ppp as far back as 2008, but it seems unlikely that this is the same problem.  (http://alastairs-place.net/blog/2007/03/12/netifaces/)

**Comment by VincentKetelaars** [comment]
Yes I read that as well. I can give you the line of code in the c program
that gave me an error ;)

PyErr_SetString (PyExc_ValueError, "You must specify a valid interface
name.");

2013/12/11 Boudewijn notifications@github.com
